frontend/setupChelonia.js

```diff
@@ -12,6 +12,43 @@ import manifests from './model/contracts/manifests.json'
 import { SETTING_CHELONIA_STATE, SETTING_CURRENT_USER } from './model/database.js'
 import { CHATROOM_USER_STOP_TYPING, CHATROOM_USER_TYPING, CHELONIA_STATE_MODIFIED, KV_EVENT, LOGGING_OUT, LOGIN_COMPLETE, LOGOUT, OFFLINE, ONLINE, RECONNECTING, RECONNECTION_FAILED, SERIOUS_ERROR } from './utils/events.js'
 
+const handleDeletedContract = async (contractID: string) => {
+  const { cheloniaState, contractState } = sbp('chelonia/contract/fullState', contractID)
+  if (!cheloniaState) return
+
+  await sbp('chelonia/contract/remove', contractID)
+
+  const type = cheloniaState.type?.replace(/^gi\.contracts\//, 'gi.actions/')
+  const handler = type && sbp('sbp/selectors/fn', `${type}/_ondeleted`)
+
+  const currentIdentityState = sbp('state/vuex/getters').currentIdentityState
+  // Delete redudant file deletion tokens. If a contract has been deleted, so
+  // have been the files it contained, and therefore we no longer need to
+  // hold on to their file deletion tokens.
+  if (currentIdentityState.fileDeleteTokens) {
+    // $FlowFixMe[incompatible-use]
+    const manifestCids = Object.entries(currentIdentityState.fileDeleteTokens).filter(([, { billableContractID }]) => {
+      return billableContractID === contractID
+    }).map(([cid]) => cid)
+    await sbp('gi.actions/identity/removeFiles', {
+      manifestCids,
+      option: {
+        shouldDeleteToken: true
+      }
```

Comment:
This seems like it will result in a call to `'chelonia/fileDelete'` which will try to delete the file on the server, but per the comment, those files should already be deleted, so this call shouldn't be made.

Perhaps either an additional option should be added here to skip that or ... what?

Reply:
> Perhaps either an additional option should be added here to skip that or ... what?

That's already done. Calling `chelonia/fileDelete` requires `shouldDeleteFile`